pybedlite/overlap_detector.py

```diff
@@ -100,26 +101,47 @@ def length(self) -> int:
 class OverlapDetector:
     """Detects and returns overlaps between a set of genomic regions and another genomic region.
 
-    If two intervals have the same coordinates, only the first that was added will be kept.
-
     Since :class:`~samwell.overlap_detector.Interval` objects are used both to populate the
     overlap detector and to query it, the coordinate system in use is also 0-based open-ended.
+
+    The same interval may be added multiple times, but only a single instance will be returned
+    when querying for overlaps.  Intervals with the same coordinates but different names are
+    treated as different intervals.
+
+    This detector is the most efficient when all intervals are added ahead of time.
     """
 
     def __init__(self) -> None:
-        self._refname_to_tree: Dict[str, IntervalTree] = {}
+        # A mapping from the contig/chromosome name to the associated interval tree
+        self._refname_to_tree: Dict[str, cr.cgranges] = {}
+        self._refname_to_indexed: Dict[str, bool] = {}
+        self._refname_to_intervals: Dict[str, List[Interval]] = {}
+        super().__init__()
```

Comment:
What is this class inheriting from that it's calling the `super()` init? On my reading it looks like it's just inheriting from object?

Reply:
Ooops holdover from the speed testing code, thank-you!!

```suggestion
```